stpipeline/core/pipeline.py

```diff
@@ -1161,7 +1161,10 @@ def run(self) -> None:
             except Exception:
                 raise
         else:
-            FILENAMES["mapped"] = input_mapping
+            # if mapping disabled, check that mapped.bam file exists
+            if not os.path.exists(FILENAMES["mapped"]):
```

Comment:
This is better placed inside `sanityCheck()`. You can do something like this:

```python
if self.disable_mapping and not os.path.exists(os.path.join(self.temp_folder, FILENAMES["mapped"])):
    error = f"Error argument '--disable-mapping' is set but {FILENAMES['mapped']} is missing in {self.temp_folder}."
    logger.error(error)
    raise RuntimeError(error)
```

so the else statement can be omitted  here.